cmdpolling/polling.go

```diff
@@ -80,41 +86,69 @@ func cmdStop(c *cli.Context) error {
 }
 
 // Main polling background routine
-func pingRoutine(ctx context.Context, c *cli.Context, pollingPingTimingInterval int64) {
+func pingRoutine(ctx context.Context, c *cli.Context, longTimePeriod int64, shortTimePeriod int64) {
 	log.Debug("pingRoutine")
 
 	formatter := format.GetFormatter()
 	pollingSvc := cmd.WireUpPolling(c)
 
+	// initialization
 	isRunningCommandRoutine := false
-	t := time.NewTicker(time.Duration(pollingPingTimingInterval) * time.Second)
+	longTicker := time.NewTicker(time.Duration(longTimePeriod) * time.Second)
+	shortTicker := time.NewTicker(time.Duration(shortTimePeriod) * time.Second)
+	currentTicker := longTicker
 	for {
-		log.Debug("Requesting for candidate commands status")
-		ping, status, err := pollingSvc.Ping()
-		if err != nil {
-			formatter.PrintError("Couldn't receive polling ping data", err)
-		} else {
-			// One command is available, and no process running
-			if status == 201 && ping.PendingCommands && !isRunningCommandRoutine {
-				log.Debug("Detected a candidate command")
-				isRunningCommandRoutine = true
-				go processingCommandRoutine(pollingSvc, formatter)
+		// no need to request until current command ends
+		if !isRunningCommandRoutine {
+			log.Debug("Requesting for candidate commands status")
+			ping, status, err := pollingSvc.Ping()
+			if err != nil {
+				formatter.PrintError("Couldn't receive polling ping data", err)
+				// in low level error, should the ticker set as the the longest time interval?
+			} else {
+				// One command is available
+				if status == 201 {
+					if ping.PendingCommands {
+						log.Debug("Detected a candidate command")
+						isRunningCommandRoutine = true
+						// set short interval timing!?
+						// - by default this implies next interval timing sil be short.
+						// - If command routine has a over delay, long interval timing will be assigned later
+						if currentTicker != shortTicker {
+							log.Debug("Ticker assigned: Short")
+							shortTicker = time.NewTicker(time.Duration(shortTimePeriod) * time.Second)
+							currentTicker = shortTicker
+						}
+						go processingCommandRoutine(pollingSvc, formatter)
+					} else {
+						// Since no pending command, long interval timing
+						log.Debug("Ticker assigned: Long")
+						currentTicker = time.NewTicker(time.Duration(longTimePeriod) * time.Second)
+					}
+				}
 			}
 		}
 
-		select {
-		case <-commandProcessed:
```

Comment:
There should be 2 selects after the ping, but you got the reversed order:
First we should check if a command ended (the one you have removed here). If so, we set the isRunningCommandRoutine to false and set the currentTicker to a new short ticker